### ISSUE TITLE
Why such a restrictive license?

### ISSUE THREAD

**Issue by nexussays** [SiliconStudio/xenko#204]
_(I couldn't find any information on your website, nor from googling, nor in other issues. My apologies if I overlooked something.)_

Why did you decide to license under GPLv3? You gain nothing from it.

Why not license under MPL 2.0 or Apache 2.0?

**Comment by xoofx** [comment]
Hi,
Thanks for your feedback. There are two types of license currently with Paradox from this [page](http://paradox3d.net/features/licensing):
- A license for binaries, that allows you to use the engine without having to pay any license fees but with the restriction that you cannot change the code of the engine
- A GPLv3 license for the source code, that allows you to modify the engine, but you either need to contribute back your work on the engine by making a pull request on github (and the changes will be integrated in the next release of the engine) or you need to release the source of your game and changes of Paradox (as stated by the GPLv3 license)

We have initially chosen this licensing model as it was the safest licensing model to protect our work (and business model), while still allowing users to contribute code back on github (without having them to release the code of their game). So if you think about it, it is not so restrictive...

That being said, we completely understand the opportunity to have a more flexible license and we have been actually considering to change it in a near future but we haven't yet settled the details.

**Comment by nexussays** [comment]
The critical problem (as I see it) is that you're fracturing the potential developer base. If I'm making a game using Paradox then the engine is, for all intents and purposes, just "source open" (as Microsoft would say) not open source. For example, if I find a bug or need a new feature in Paradox, I am 100% dependent on you accepting that change upstream in order to not be infected by the GPL. I can't imagine any game developer who would risk that.

If you licensed under Apache or MPL 2.0 then you get the best of both worlds -- no one can make private changes to Paradox and game developers aren't forced to open source their product.

Wouldn't that better fit with your goals?

**Comment by xoofx** [comment]
> Wouldn't that better fit with your goals?

Yep, that could be, as I said, It is part of our reflection to move to a more flexible license ;)

**Comment by nexussays** [comment]
Good to hear! If/when you decide to change the license be sure to mention it in this issue :)

**Comment by FREEZX** [comment]
Ok, so from the FAQ:

```
Can I sell games made with Xenko Beta?

Yes, you can sell your game without any constraints as long as you use the binary version distributed on the website. If you modify and recompile the engine yourself, you can still sell your product, but you have to open the source code of your game and the modifications made to Xenko as stated in the GPL license.
```

This seems too restrictive for me to decide to start with Xenko. Even if i do some development on an issue i may be having in the engine and open source it, unless the maintainers decide to accept my PR and distribute new binaries, i'll be screwed by having to release my full game source code and assets under GPLv3.

That said, if i was able to NOT open source my game, but only open source my work on the engine, i'd gladly switch and contribute.

**Comment by Kryptos-FR** [comment]
You don't have to release your assets under GPL.
